source/slang/slang-check-expr.cpp

```diff
@@ -2376,6 +2376,13 @@ namespace Slang
         LogicOperatorShortCircuitExpr* newExpr = nullptr;
         isArgumentsChecked = false;
 
+        // If the logic expression is inside the generic parameter list, it cannot support short-circuit
+        // which will generate the ifelse branch.
+        if (m_outerGenericApp != nullptr || m_outerDecl != nullptr)
```

Comment:
Another case we need to detect is `int array[a&&b]`.

Reply:
Probably instead of keeping m_outerGenericApp or m_outerDecl as state, just keep a `bool shouldShortCircuitLogicalExpr` as the state.

Reply:
Yea, I'm also considering this. Because in those circumstances, there should not be nested?

Reply:
Update using `bool shouldShortCircuitLogicalExpr` : https://github.com/shader-slang/slang/pull/3635/commits/4759154ba75b2d59ab77232bb3471bf409143d08

Disable short-circuit in index expression: https://github.com/shader-slang/slang/pull/3635/commits/2684bc5812fc217653afe3bf3d19fadb72555b02